ext/pdo/pdo_stmt.c

```diff
@@ -1755,120 +1782,139 @@ int pdo_stmt_setup_fetch_mode(pdo_stmt_t *stmt, zend_long mode, zval *args, uint
 		case PDO_FETCH_BOUND:
 		case PDO_FETCH_NAMED:
 		case PDO_FETCH_KEY_PAIR:
-			if (num_args != 0) {
-				pdo_raise_impl_error(stmt->dbh, stmt, "HY000", "fetch mode doesn't allow any extra arguments");
-			} else {
-				retval = SUCCESS;
+			if (total_num_args != mode_arg_num) {
```

Comment:
The previous comparison made more sense? `num_args != 0` is pretty obvious, `total_num_args != mode_arg_num` is not. Same for the other argument number comparisons.

Reply:
I'll revert them, because that's true. I was just trying to get it to work ASAP (and it still needs some work in regards to the compiler complaining for good reasons)